crates/deltalake-core/src/operations/vacuum.rs

```diff
@@ -195,12 +195,48 @@ impl VacuumBuilder {
             .ok_or(DeltaTableError::NoMetadata)?
             .partition_columns;
 
+        let managed_files = self
+            .snapshot
+            .files()
+            .iter()
+            .map(|a| a.path.clone())
+            .chain(
+                self.snapshot
+                    .all_tombstones()
+                    .iter()
+                    .map(|r| r.path.clone()),
+            )
+            .chain(self.snapshot.files().iter().filter_map(|a| {
+                return if let Some(deletion_vector) = &a.deletion_vector {
+                    if let Ok(parent) = &Url::parse(self.log_store.root_uri().as_str()) {
+                        if let Ok(dv_absolut_path) = deletion_vector.absolute_path(&parent) {
+                            Some(dv_absolut_path?.to_string())
+                        } else {
+                            None
+                        }
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                };
+            }))
+            .collect::<HashSet<String>>();
```

Comment:
Since the deletion vector file path is being constructed on the fly by the `absolut_path` function, it now must be a `HashSet<String>` instead of `HashSet<&str>`. This also means that we now have to clone the String of the Add and Remove path. Is this fine or is there a better way to handle this?

Also, this nested handling of the deletion vectors doesn't look nice, can you show me a better/more idiomatic way to achieve this?